### PR TITLE
validate restaurant email and password with login/registration 

### PR DESCRIPTION
**Pull request by audrius-savickas**
Restaurants can now register and login into their accounts

### CODE REVIEW

File: wasted-app/RestaurantRegistrationControl.cs [hunk 1]
```diff
@@ -38,15 +40,20 @@ private void registerButton_Click(object sender, EventArgs e)
...
-            if (error == "")
+            String credentialError = getValidationError(username, password);
+
+            if (credentialError == "")
             {
-                passwordError.Text = "";
+                this.passwordError.Text = "";
+                ServicesController controller = ServicesController.Instance;
+                Credentials creds = new Credentials(username, password);
+                controller.RestaurantService.Register(creds, new Restaurant("aa", "Hello World", new Coords(10, 20), new Credentials()));
```
Comment (audrius-savickas) [inline wasted-app/RestaurantRegistrationControl.cs:50]: should add proper id, not "aa".
should add a new field for the restaurant name, not "Hello World".

File: wasted-app/Controller/Services/RestaurantService.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public IEnumerable<Restaurant> GetRestaurantsNear(Coords coords)
         public bool Login(Credentials creds)
         {
             Restaurant restaurant = GetByMail(creds.Mail);
-            return restaurant.Credentials.Equals(creds);
+            return restaurant != null && restaurant.Credentials.Mail.Value == creds.Mail.Value && restaurant.Credentials.Password.Value == creds.Password.Value;
```
Comment (audrius-savickas) [inline wasted-app/Controller/Services/RestaurantService.cs:56]: move to a separate function?

Comment (DonatasPuc) [reply wasted-app/Controller/Services/RestaurantService.cs:56]: I don't think so. Looks good to me.

File: wasted-app/RestaurantRegistrationControl.cs (hunk 1 above)
Comment (JesusGonzalezA) [reply wasted-app/RestaurantRegistrationControl.cs:50]: When we'll change to databases the id field will be generated automatically

File: wasted-app/Controller/Services/RestaurantService.cs (hunk 2 above)
Comment (JesusGonzalezA) [reply wasted-app/Controller/Services/RestaurantService.cs:56]: For me as well

Comment (DominykasSip) [reply wasted-app/Controller/Services/RestaurantService.cs:56]: Looks fine to me.

**Pull request merged**